python/cudf/cudf/core/dataframe.py

```diff
@@ -5288,39 +5294,62 @@ def quantile(
         if columns is None:
             columns = data_df._data.names
 
-        # Ensure that qs is non-scalar so that we always get a column back.
-        qs = [q] if is_scalar(q) else q
-        result = {}
-        for k in data_df._data.names:
-            if k in columns:
-                ser = data_df[k]
-                res = ser.quantile(
-                    qs,
-                    interpolation=interpolation,
-                    exact=exact,
-                    quant_index=False,
-                )._column
-                if len(res) == 0:
-                    res = column.column_empty_like(
-                        qs, dtype=ser.dtype, masked=True, newsize=len(qs)
-                    )
-                result[k] = res
+        if isinstance(q, numbers.Number):
+            q_is_number = True
+            qs = [float(q)]
+        elif pd.api.types.is_list_like(q):
+            q_is_number = False
+            qs = q
+        else:
+            msg = "`q` must be either a single element or list"
+            raise TypeError(msg)
 
-        result = DataFrame._from_data(result)
-        if isinstance(q, numbers.Number) and numeric_only:
-            result = result.fillna(np.nan).iloc[0]
-            result.index = data_df._data.to_pandas_index()
-            result.name = q
-            return result
+        if method == "table":
+            interpolation = interpolation or "nearest"
+            result = self._quantile_table(qs, interpolation.upper())
+
+            if q_is_number:
+                result = result.transpose()
+                return Series(
+                    data=result._columns[0], index=result.index, name=q
```

Comment:
This produces a slight inconsistency with the other cases in that here the index is a `cudf.core.index.StringIndex` whereas otherwise it is a `pandas.core.indexes.base.Index`.

Reply:
Does [this change](https://github.com/rapidsai/cudf/pull/11957#discussion_r1003329634) resolve the inconsistency? Side note that this isn't new code, but I'm certainly happy to fix things while we are making changes anyway.